libsrc/supervision/crt0.s

```diff
@@ -67,7 +67,10 @@ not_dma:
 ; Removing this segment gives only a warning.
         .segment "FFF0"
 .proc reset32kcode
-        lda     #(6<<5)
+        lda     #$A0
```

Comment:
I don't see why initializing the screen should happen in a function called reset32kcode. This somehow looks like a hack rather than a fix.

In general I don't see why this code should be placed in crt0.s. This target doesn't have any CONIO support so why should it initialize the screen in the first place. Without CONIO support it's _all_ up to the user. And if there's CONIO support at some point then the screen initialization belongs to a CONIO constructor.

Reply:
Ok, but I am not sure about your point: what is the of not having the screen on and the resolution set up to some useful default value? This is not a conio init. This is just a very generic initialization that has nothing to do with conio, cursors, text, etc.
Let us at least have one helloworld example for this under-documented target 
I can move the screen init part into the hello world example and I improve the example if necessary.

Reply:
@oliverschmidt So, is it ok to have the screen init code in crt0.s or should I just mobe it to the helloworld example in a separate PR? I do as you prefer.

Reply:
The bank setting is needed, in "crt0.s", in order to make cartridges work in a predictable way.  I think that the other parts of that register may as well be set at the same time.  But, the screen size settings should be with code that actually uses the screen.  At this time, it's the "Hello world" program.  (Later on, we might have an implementation of conio.  Then, it would set those sizes.)

Reply:
@greg-king5: That's exactly my perspective too. Thanks for elaborating.

Reply:
Ok! So I will revert my changes about lcd size in the cfg and crt0.s.

Reply:
@oliverschmidt  should I cancel this PR and make two PRs? One for crt and one for the example?
Should I keep this one for the crt?

Reply:
@oliverschmidt @greg-king5  the reason why I put the 160x160 setting is that this tells the controller it is on a 160x160 which is the Watara Supervision one. 160x160 is going to be always used for the Watara Supervision target. The video control can be told to use different screens but this is never the case on this handheld. I cannot think of a real use case with different values. Not having this setting makes the crt a generic non-Supervision one. Is this what we really want?

Reply:
Supervision display is always 160x160 and the crt0.s for the Supervision can assume it is running on a Supervision which only has a fixed size screen. No other values will ever be used.
What is the use-case you guys are thinking of?

Reply:
I agree, Fabri. By the way isn't it natural to use `#160` if you say 160 pixel?

Reply:
@polluks $A is 10 (just after 9) and second hex position is multiples of 16... So $A0 = 160 is quite obvious to me.

Reply:
@oliverschmidt do you really want me to remove the lcd size setting from the crt even though this is the only meaningful value for the Supervision taget in all (reasonable) use-cases?
Other lcd size values make sense for other screen in other consoles with the same video controller.
I am just asking you to be sure. I will not insist on this.

Reply:
> @polluks, $A is 10 (just after 9), and the second hex position is multiples of 16... So, $A0 = 160 is quite obvious to me.

It isn't obvious to me -- I need to do the arithmetic in my head every time that I see it.

Reply:
@greg-king5 16x10 is just adding a zero but OK, I am going to make it even clearer.

Reply:
@greg-king5 I have change it to #160 but the real problem is whether it should be done or not... I am doing it in crt for the reasons that I have given. LCD on Supervision is fixed to 160x160 and other values have no (real) use on the Supervision no matter what you use (conio, low level, tgi, stdio, etc.).
I will remove it from there if @oliverschmidt decides so.